Panic in nodeAgent

**Describe the bug**:
After upgrade from 3.12 to 3.17 I encounter a panic when starting logging-operator. Using the helm chart to deploy the operator.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x162c7e2]

goroutine 732 [running]:
github.com/banzaicloud/logging-operator/pkg/resources/nodeagent.(*nodeAgentInstance).configSecret(0xc0016d6090, 0x1be7798, 0xc0002f0f20, 0x1c0e570, 0x1bc4140, 0x0, 0x0)
        /workspace/pkg/resources/nodeagent/configsecret.go:237 +0x8a2
github.com/banzaicloud/logging-operator/pkg/resources/nodeagent.(*nodeAgentInstance).Reconcile(0xc0016d6090, 0xc000fb8b00, 0x170d980, 0xc000c1c258)
        /workspace/pkg/resources/nodeagent/nodeagent.go:351 +0x310
github.com/banzaicloud/logging-operator/pkg/resources/nodeagent.(*Reconciler).Reconcile(0xc0012d3a10, 0x0, 0x0, 0x0)
        /workspace/pkg/resources/nodeagent/nodeagent.go:325 +0x215
github.com/banzaicloud/logging-operator/controllers/logging.(*LoggingReconciler).Reconcile(0xc0003e9240, 0x1c110d8, 0xc00012fb30, 0x0, 0x0, 0xc000c47788, 0x18, 0xc00012fb30, 0xc000030000, 0x188b4a0, ...)
        /workspace/controllers/logging/logging_controller.go:128 +0x605
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002a9220, 0x1c11030, 0xc0002ee000, 0x184b3e0, 0xc000bb1ec0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002a9220, 0x1c11030, 0xc0002ee000, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc0002a6200, 0xc0002a9220, 0x1c11030, 0xc0002ee000)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:210 +0x425
```

**Expected behaviour**:
Panic does not happen.

**Steps to reproduce the bug**:
Config:
```yaml
# values of logging operator
nodeSelector:
  node.helio.exchange/role: "worker"
  kubernetes.io/os: "linux"

replicaCount: 1
```

```yaml
# logging CRD config
# Fluent-bit configurations https://banzaicloud.com/docs/one-eye/logging-operator/configuration/crds/#fluent-bit-spec
# fluentbit:
# tolerations:
#   - operator: Exists
# image:
#   repository: eu.gcr.io/k8s-helio-development/fluent/fluent-bit
#   tag: 1.7.2

# If an immutable field is changed, delete the existing resource
# and recreate it with the new configuration.
enableRecreateWorkloadOnImmutableFieldChange: true

nodeAgents:
  - name: winagent
    profile: windows
    nodeAgentFluentbit:
      tls:
        enabled: false
      daemonSet:
        spec:
          template:
            spec:
              nodeSelector:
                kubernetes.io/os: windows
              tolerations:
                - key: "node.kubernetes.io/os"
                  operator: Equal
                  value: "windows"
                  effect: NoSchedule
                - operator: Exists
              containers:
                - name: fluent-bit
                  image: REDACTED/fluent-bit:1.7.9
                  resources:
                    limits:
                      cpu: 500m
                      memory: 500M
                    requests:
                      cpu: 200m
                      memory: 200M
  - name: linuxagent
    profile: linux
    nodeAgentFluentbit:
      tls:
        enabled: false
      daemonSet:
        spec:
          template:
            spec:
              nodeSelector:
                kubernetes.io/os: linux
              tolerations:
                - operator: Exists
              containers:
                - name: fluent-bit
                  image: REDACTED/fluent-bit:1.7.9
                  livenessProbe:
                    httpGet:
                      path: /
                      port: 2020
                    initialDelaySeconds: 10
                    periodSeconds: 10
                    successThreshold: 1
                    failureThreshold: 3
                    timeoutSeconds: 10

# Fluentd configurations https://banzaicloud.com/docs/one-eye/logging-operator/configuration/crds/#fluentd-spec
fluentd:
  image:
    tag: v1.13.1-alpine-2
  livenessDefaultCheck: true
  security:
    securityContext:
      allowPrivilegeEscalation: false
      readOnlyRootFilesystem: false
    podSecurityContext:
      # seemingly /buffers is always root:root, even if we specify 101 as the group. no idea why this happens,
      # the docker image already creates /buffers using root:fluent (or so) but k8s should chown anyway but doesn't.
      fsGroup: 0
      runAsUser: 0
      runAsGroup: 0
  podPriorityClassName: "system-cluster-critical"
  bufferStorageVolume:
    pvc:
      spec:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 32Gi
        storageClassName: managed-premium
        volumeMode: Filesystem
  scaling:
    replicas: 2
  resources:
    limits:
      cpu: "1"
      memory: 2Gi
    requests:
      cpu: "1"
      memory: 1Gi
  nodeSelector:
    kubernetes.io/role: "master"
    kubernetes.io/os: "linux"
  tolerations: REDACTED

# Enable secure connection between fluentd and fluent-bit
tls:
  enabled: false
  # Shared key for fluentd authentication
  sharedKey: ""
  fluentbitSecretName: ""
  fluentdSecretName: ""

# Control namespace that contains ClusterOutput and ClusterFlow resources
controlNamespace: "logging"

# ClusterFlows to deploy
clusterFlows:
  - name: all
    spec:
      filters:
        - tag_normaliser:
            format: "${namespace_name}.${pod_name}.${container_name}"
        - parser:
            key_name: log
            reserve_data: true
            remove_key_name_field: true
            parse:
              type: multi_format
              patterns:
                - format: json
                  time_key: ""
                - format: regexp
                  expression: ^t(ime)?=(?<time>[^\s]*) le?ve?l=(?<level>[a-z]+) msg="(?<message>.+)"( ([a-z]+)=(?:(?:"(.*)")|(?:(?:([^\s]+)[\s]))).*)?$
                  time_key: "time"
                - format: none
        - parser:
            key_name: message
            reserve_data: true
            emit_invalid_record_to_error: false
            parse:
              type: regexp
              expression: ^t(ime)?=(?<time>[^\s]*) le?ve?l=(?<level>[a-z]+) msg="(?<message>.+)"( ([a-z]+)=(?:(?:"(.*)")|(?:(?:([^\s]+)[\s]))).*)?$
              time_key: "time"
        - record_transformer:
            remove_keys: $.kubernetes.pod_id,$.kubernetes.container_hash,$.kubernetes.docker_id,$.time,$.kubernetes.labels.pod-template-hash,$.kubernetes.annotations.checksum/fluent-bit-conf.type,$.kubernetes.annotations.cluster-autoscaler-kubernetes-io/daemonset-pod.type,$.kubernetes.annotations.kubernetes-io/config-hash.type,$.kubernetes.annotations.kubernetes-io/config-mirror.type,$.kubernetes.annotations.kubernetes-io/config-seen.type,$.kubernetes.annotations.kubernetes-io/config-source.type,$.kubernetes.annotations.kubernetes-io/psp.type,$.kubernetes.annotations.scheduler-alpha-kubernetes-io/critical-pod.type,$.kubernetes.annotations.seccomp-security-alpha-kubernetes-io/pod.type
        - record_modifier:
            records:
              - cluster: REDACTED
        - dedot:
            de_dot_separator: "-"
            de_dot_nested: true
      globalOutputRefs:
        - elasticsearch

# ClusterOutputs to deploy
clusterOutputs:
  - name: elasticsearch
    spec:
      elasticsearch:
        host: REDACTED
        port: REDACTED
        user: REDACTED
        password:
          valueFrom:
            secretKeyRef:
              name: es-password
              key: password
        scheme: https
        ssl_verify: true
        include_timestamp: true
        reload_on_failure: true
        reload_connections: false
        buffer:
          flush_mode: interval
          flush_interval: 60s
          flush_at_shutdown: true
          flush_thread_count: 4
          overflow_action: throw_exception
        index_name: REDACTED
        log_es_400_reason: true
        suppress_type_name: true
        data_stream_enable: true
        data_stream_name: REDACTED
```

**Additional context**:
I'm aware that I have some custom image tags set - will check those. I currently believe that's independent from the error though. 

**Environment details**:
- Kubernetes version (e.g. v1.15.2): v1.21.4
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): custom AKS
- logging-operator version (e.g. 2.1.1): 3.17.0
- Install method (e.g. helm or static manifests): helm

```
{"level":"info","ts":1643098428.65353,"msg":"Waited for 1.029403643s due to client-side throttling, not priority and fairness, request: GET:https://10.0.0.1:443/apis/discovery.k8s.io/v1beta1?timeout=32s\n"}
{"level":"info","ts":1643098429.0596716,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1643098429.064281,"logger":"setup","msg":"Detected container runtime","runtime":"docker"}
{"level":"info","ts":1643098429.0643122,"logger":"setup","msg":"Unable to detect container runtime, keeping default value","runtime":"docker"}
{"level":"info","ts":1643098429.064496,"logger":"setup","msg":"starting manager"}
{"level":"info","ts":1643098429.0647523,"msg":"attempting to acquire leader lease logging/logging-operator.logging.banzaicloud.io...\n"}
{"level":"info","ts":1643098429.0648324,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"info","ts":1643098446.6741624,"msg":"successfully acquired lease logging/logging-operator.logging.banzaicloud.io\n"}
{"level":"info","ts":1643098446.6743982,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744366,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744463,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744523,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744592,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744654,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744733,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744788,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744847,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674492,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674502,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6745074,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6745136,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6744733,"logger":"controller-runtime.manager.controller.eventtailer","msg":"Starting EventSource","reconciler group":"logging-extensions.banzaicloud.io","reconciler kind":"EventTailer","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6745355,"logger":"controller-runtime.manager.controller.eventtailer","msg":"Starting Controller","reconciler group":"logging-extensions.banzaicloud.io","reconciler kind":"EventTailer"}
{"level":"info","ts":1643098446.6745534,"logger":"controller-runtime.manager.controller.hosttailer","msg":"Starting EventSource","reconciler group":"logging-extensions.banzaicloud.io","reconciler kind":"HostTailer","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674609,"logger":"controller-runtime.manager.controller.hosttailer","msg":"Starting Controller","reconciler group":"logging-extensions.banzaicloud.io","reconciler kind":"HostTailer"}
{"level":"info","ts":1643098446.674521,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674639,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6746943,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674703,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.67471,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6747174,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6747246,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6747324,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6747396,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6747475,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.6747577,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674779,"logger":"controller-runtime.manager.controller.logging","msg":"Starting EventSource","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","source":"kind source: /, Kind="}
{"level":"info","ts":1643098446.674796,"logger":"controller-runtime.manager.controller.logging","msg":"Starting Controller","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging"}
{"level":"info","ts":1643098446.7758243,"logger":"controller-runtime.manager.controller.hosttailer","msg":"Starting workers","reconciler group":"logging-extensions.banzaicloud.io","reconciler kind":"HostTailer","worker count":1}
{"level":"info","ts":1643098446.777413,"logger":"controller-runtime.manager.controller.eventtailer","msg":"Starting workers","reconciler group":"logging-extensions.banzaicloud.io","reconciler kind":"EventTailer","worker count":1}
{"level":"info","ts":1643098446.7782092,"logger":"controller-runtime.manager.controller.logging","msg":"Starting workers","reconciler group":"logging.banzaicloud.io","reconciler kind":"Logging","worker count":1}
{"level":"info","ts":1643098447.093833,"logger":"KubeAPIWarningLogger","msg":"policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+"}
{"level":"info","ts":1643098447.409383,"logger":"controllers.Logging","msg":"fluentd buffer draining is disabled"}
{"level":"info","ts":1643098447.411878,"msg":"Notice: If the Buffer_Size value is empty we will set it 0. For more information: https://github.com/fluent/fluent-bit/issues/2111"}
{"level":"info","ts":1643098447.41199,"msg":"Notice: Because the Fluentd statefulset has been scaled, we've made some changes in the fluentbit network config too. We advice to revise these default configurations."}
{"level":"info","ts":1643098447.4236503,"msg":"Notice: If the Buffer_Size value is empty we will set it 0. For more information: https://github.com/fluent/fluent-bit/issues/2111"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x162c7e2]

goroutine 669 [running]:
github.com/banzaicloud/logging-operator/pkg/resources/nodeagent.(*nodeAgentInstance).configSecret(0xc0010f48a0, 0x1be7798, 0xc001497340, 0x1c0e570, 0x1bc4140, 0x0, 0x0)
        /workspace/pkg/resources/nodeagent/configsecret.go:237 +0x8a2
github.com/banzaicloud/logging-operator/pkg/resources/nodeagent.(*nodeAgentInstance).Reconcile(0xc0010f48a0, 0xc000f8b440, 0x170d980, 0xc000656458)
        /workspace/pkg/resources/nodeagent/nodeagent.go:351 +0x310
github.com/banzaicloud/logging-operator/pkg/resources/nodeagent.(*Reconciler).Reconcile(0xc00113c6f0, 0x0, 0x0, 0x0)
        /workspace/pkg/resources/nodeagent/nodeagent.go:325 +0x215
github.com/banzaicloud/logging-operator/controllers/logging.(*LoggingReconciler).Reconcile(0xc000470f60, 0x1c110d8, 0xc000fd26f0, 0x0, 0x0, 0xc0005d4630, 0x18, 0xc000fd26f0, 0xc000030000, 0x188b4a0, ...)
        /workspace/controllers/logging/logging_controller.go:128 +0x605
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00018ef00, 0x1c11030, 0xc0006e1b40, 0x184b3e0, 0xc000d76080)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00018ef00, 0x1c11030, 0xc0006e1b40, 0xc0004c6700)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc0005b3850, 0xc00018ef00, 0x1c11030, 0xc0006e1b40)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:210 +0x425
```

/kind bug

--- a/pkg/resources/nodeagent/configsecret.go
+++ b/pkg/resources/nodeagent/configsecret.go
@@ -234,6 +234,10 @@ func (n *nodeAgentInstance) configSecret() (runtime.Object, reconciler.DesiredSt
 		}
 	}
 
+	if nil == n.fluentdDataProvider {
+		return nil, nil, errors.WrapIf(err, "nil fluent data provider")
+	}
+
 	fluentdReplicas, err := n.fluentdDataProvider.GetReplicaCount(context.TODO(), n.logging)
 	if err != nil {
 		return nil, nil, errors.WrapIf(err, "getting replica count for fluentd")

--- a/pkg/resources/nodeagent/nodeagent.go
+++ b/pkg/resources/nodeagent/nodeagent.go
@@ -317,9 +317,10 @@ func (r *Reconciler) Reconcile() (*reconcile.Result, error) {
 		}
 
 		instance = nodeAgentInstance{
-			nodeAgent:  NodeAgentFluentbitDefaults,
-			reconciler: r.GenericResourceReconciler,
-			logging:    r.Logging,
+			nodeAgent:           NodeAgentFluentbitDefaults,
+			reconciler:          r.GenericResourceReconciler,
+			logging:             r.Logging,
+			fluentdDataProvider: r.fluentdDataProvider,
 		}
 
 		result, err := instance.Reconcile()

